### PR TITLE
blog: revamp "Switch from Docker Desktop to Rancher Desktop"

### PR DESCRIPTION
**Pull request by milas**
Update based on ecosystem changes since originally published in
Sept 2021. (If you're keeping score, Nick predicted this right
away.)

The original post, which is focused on the now inactive `kim`
builder experiment that shipped with early versions of Rancher
Desktop is now moved to a "new" (but backdated) post for posterity,
as it's not a bad example of writing a `custom_build` extension.
It also links to the "Switch to Rancher Desktop" post for any
Rancher Desktop users that stumble upon it, as it's now no longer
relevant for them.

The new post is pretty short and sweet because there's not a ton
to do! But it lives at the old slug (SEO yay) and _really_ fulfills
its promise of a 5 minute swap :)

See also tilt-dev/tilt#5513, which adds the in-cluster detection
support, so is required for this to work as expected.

### CODE REVIEW

**Comment by milas** [comment]
This is part 1 of my Rancher Desktop content 🌊  - currently working on an (actually) new post featuring the `nerdctl` extension (tilt-dev/tilt-extensions#342) for if you want to use Rancher Desktop in containerd mode (default).

**Comment by nicks** [comment]
wait - i'm a little confused. i would expect we want a new post dated 2022, but to keep the old post mostly the same (with an update banner)? i had trouble following what's going in in this pr, but it seems like we're splitting into two posts dated september 2021?

**Comment by milas** [comment]
Both - I'm writing a new post! Which will be more focused on containerd vs dockerd (wrt Rancher Desktop + Tilt) and why you might pick each / how you'd use it in that case, i.e. the new `nerdctl_build` extension for containerd or "change the setting and upgrade Tilt" for dockerd, which will mostly be a pointer to the "Switch in 5 Minutes" post.

My thought process:
* Old post is confusingly irrelevant now that `kim` isn't bundled with / mentioned anywhere in RD docs anymore
* Old post gets a lot of traffic (yay snappy title) and "switch to dockerd and it just works" actually does fulfill the "5 min" promise
* New post will be a bit more of a "container runtime + intro to `nerdctl_build`" post (w/ a link to the revamped "Switch in 5 Minutes" post for those who just want to get things working ASAP)

IMO that gives a good balance of new content while ensuring the most likely place users end up ("Switch in 5 Mins" post) has the info they need w/o confusing anyone about `kim`

But I can also just add the new post and a disclaimer+link on the old one if preferable! 🙃

File: blog/_posts/2021-09-05-kim-custom-build-extension.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+image_caption: '<a href="https://www.flickr.com/photos/44124427152@N01/42670013">"construction in Dublin Docklands"</a> by Salim Virji (<a href="https://creativecommons.org/licenses/by-sa/2.0/">CC BY-SA 2.0</a>)'
+description: "Configure Tilt to build images with the experimental Kubernetes Image Manager (kim) project"
+tags:
+- docker
+- kubernetes
+---
+
+---
+
+## **UPDATE**: The `kim` project is not currently active
```
[the remark this answers is not in the document]
Comment (lianmakesthings) [reply blog/_posts/2021-09-05-kim-custom-build-extension.md:17]:
```suggestion
## **UPDATE February 2022**: The `kim` project is not currently active
```

**Review by lianmakesthings — APPROVED** [review]

**Review by nicks — APPROVED** [review]

**Comment by nicks** [comment]
i really liked the framing of "if you want a drop-in replacement, configure it to use docker.

I might add a little note about a future tilt extension if you want to run it with containerd.

**Comment by milas** [comment]
> I might add a little teaser about a future blog post + tilt extension if you want to run it with containerd.

Good call! I put that in [19688fe](https://github.com/tilt-dev/tilt.build/pull/1070/commits/19688fe06b7fcb21b5c2496562573b4561826c15) (part of #1070, which is the new post itself)

**Pull request merged**